plugins/contentful/src/plugin.ts

```diff
@@ -0,0 +1,143 @@
+              ['Text', 'Boolean', 'Number', 'Symbol'].includes(field.type)
+            );
+            const fields: any = [
+              {
+                name: 'include',
+                friendlyName: 'limit',
+                defaultValue: 10,
+                // contentful api restricts include to be between 0 and 10
+                min: 0,
+                max: 10,
+                type: 'number',
+              },
+              {
+                name: 'order',
+                type: 'string',
+                enum: Object.keys(type.sys)
+                  .filter(key => !metaFields.includes(key))
+                  .map(key => ({
+                    label: key,
+                    value: `sys.${key}`,
+                  }))
+                  .concat(
+                    acceptableFields
+                      .filter(field => field.type === 'Symbol')
+                      .map(field => ({
+                        label: field.name,
+                        value: `fields.${field.id}`,
+                      }))
+                  ),
+              },
+            ];
+
+            if (acceptableFields.length > 0) {
+              fields.push({
+                name: 'fields',
+                advanced: true,
+                type: 'object',
+                friendlyName: `${type.name} fields`,
+                subFields: acceptableFields.map(field => ({
+                  type: field.type === 'Symbol' ? 'text' : field.type.toLowerCase(),
+                  name: field.id,
+                  friendlyName: field.name,
+                  helperText: `Query by a specific "${field.name}"" on ${type.name}`,
+                })),
+              } as any);
+            }
+
+            return fields;
+          },
+          toUrl: (options: any) => {
+            // by entry
+            // https://cdn.contentful.com/spaces/{space_id}/environments/{environment_id}/entries/{entry_id}?access_token={access_token}
+            if (options.entry) {
+              // todo: maybe environment should be an input
+              return `https://cdn.contentful.com/spaces/${spaceId}/environments/master/entries/${options.entry}?access_token=${accessToken}`;
+            }
+            let fields =
+              (options.fields && Object.keys(options.fields).length > 0 && options.fields) || null;
+            if (fields) {
+              fields = Object.keys(fields).reduce((acc, key) => {
+                const omitValue = fields[key] === '';
+                return {
+                  ...acc,
+                  ...(omitValue ? {} : { [key]: fields[key] }),
+                };
+              }, {});
+            }
+
+            const params = qs.stringify(
+              { ...options, fields, access_token: accessToken, content_type: type.sys.id },
+              { allowDots: true, skipNulls: true }
+            );
+            // by query
+            return `https://cdn.contentful.com/spaces/${spaceId}/environments/master/entries?${params}`;
```

Comment:
small detail: I'd extract the common `https://cdn.contentful.com/spaces/${spaceId}/environments/master/entries/` between this and the `optoins.entry` use-case into a variable